iac/bicep/role.bicep

```diff
@@ -0,0 +1,12 @@
+param roleId string = ''
+param principalType string = 'ServicePrincipal'
+param principalId string = ''
+
+resource role_assignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(subscription().id, principalId, roleId)
```

Comment:
The `guid()` will take n arguments, so we can keep making the seed more complex. I wonder if we can make this work by adding more arguments to the `guid()` function like a resource ID or the resourceGroup ID if applicable.

Reply:
Added rgName